providers/provider_default.go

```diff
@@ -95,34 +101,36 @@ func (p *ProviderData) GetLoginURL(redirectURI, state string) string {
 }
 
 // GetEmailAddress returns the Account email address
-func (p *ProviderData) GetEmailAddress(ctx context.Context, s *sessions.SessionState) (string, error) {
-	return "", errors.New("not implemented")
-}
-
-// GetUserName returns the Account username
-func (p *ProviderData) GetUserName(ctx context.Context, s *sessions.SessionState) (string, error) {
-	return "", errors.New("not implemented")
+// DEPRECATED: Migrate to EnrichSessionState
+func (p *ProviderData) GetEmailAddress(_ context.Context, _ *sessions.SessionState) (string, error) {
```

Comment:
Does this PR remove all use of `GetEmailAddress`? Do we need to keep this in?

Reply:
Still a bunch of `GetEmailAddress` uses unfortunately.  There were only 2 `GetUserName` uses, so I refactored those now while still keeping the PR small.

My goal is to go `GetEmailAddress` over time and have the few authorization related PRs for Keycloak and Gitlab coming in clean this up as a start.